pygments/lexers/qlik.py

```diff
@@ -0,0 +1,142 @@
+"""
+    pygments.lexers.qlik
+    ~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for the qlik scripting language
+
+    :copyright: Copyright 2006-2021 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import re
+
+from pygments.lexer import RegexLexer, include, bygroups, words
+from pygments.token import *
+from pygments.lexers._qlik_builtins import (
+    OPERATORS_LIST,
+    STATEMENT_LIST,
+    SCRIPT_FUNCTIONS,
+    CONSTANT_LIST,
+)
+
+__all__ = ["QlikLexer"]
+
+field_name_options = r"(\w|\[[\w\s]\]|\"[\w\s]\")"
+
+
+class QlikLexer(RegexLexer):
+    """
+    Lexer for qlik code, including .qvs files
+    """
+
+    name = "Qlik"
+    aliases = ["qlik", "qlikview", "qliksense", "qlikscript"]
+    filenames = ["*.qvs", "*.qvw"]
+
+    flags = re.IGNORECASE
+
+    tokens = {
+        # Handle multi-line comments
+        "comment": [
+            (r"\*/", Comment.Multiline, "#pop"),
+            (r"[^\*]+", Comment.Multiline),
+        ],
+        # Handle numbers
+        "numerics": [
+            (r"\b\d+\.\d+(e\d+)?[fd]?\b", Number.Float),
+            (r"\b\d+\b", Number.Integer),
+        ],
+        # Handle variable names in things
+        "interp": [
+            (
+                r"(\$\()(\w+)(\))",
+                bygroups(String.Interpol, Name.Variable, String.Interpol),
+            ),
+        ],
+        # Handle strings
+        "string": [
+            (
+                r"'",
+                String,
+                "#pop",
+            ),
+            include("interp"),
+            (r"[^'(\$\()]+", String),
```

Comment:
The open parenthesis is repeated in the character class. Also, no need to escape dollars or parentheses (or the star in the comment state above), they lose their special meaning within the brackets of a character class.

Reply:
@Jean-Abou-Samra I'm struggling a bit to work out this one (and the two below). The idea in Qlik is that one can have e.g.

```
'String $(Variable) String $(AnotherVariable) String (Still String) $ String'
```

All except the `$(Variable)` and `$(AnotherVariable)` should be Strings. So basically I want to say "As long as it's not `$(` together, then it's a string".

Originally, not having the `$(` in this section worked on the first `$(Variable)` part, but not the second. By adding them in, it then won't allow `$`, `(` or `)` in the string.